src/main/java/org/broadinstitute/hellbender/tools/copynumber/allelic/alleliccount/AllelicCount.java

```diff
@@ -0,0 +1,98 @@
+ *
+ * @author Samuel Lee &lt;slee@broadinstitute.org&gt;
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public class AllelicCount implements Locatable {
+
+    /* these are mandatory */
+    private final SimpleInterval interval;
+    private final int refReadCount;
+    private final int altReadCount;
+    /* these are extra metadata and can be null */
+    private final Nucleotide refNucleotide;
+    private final Nucleotide altNucleotide;
+
+    /**
+     * Construct the allelic count object.
+     * @param interval the genomic interval (it is assumed to have the same begin and end points)
+     * @param refReadCount ref nucleotide count in the pileup
+     * @param altReadCount alt nucleotide count in the pileup
+     * @param refNucleotide ref nucleotide
+     * @param altNucleotide alt nucleotide
+     */
+    public AllelicCount(final SimpleInterval interval, final int refReadCount, final int altReadCount,
+                        final Nucleotide refNucleotide, final Nucleotide altNucleotide) {
+        this.interval = Utils.nonNull(interval, "Can't construct AllelicCount with null interval.");
+        this.refReadCount = ParamUtils.isPositiveOrZero(refReadCount, "Can't construct AllelicCount with negative read counts.");
+        this.altReadCount = ParamUtils.isPositiveOrZero(altReadCount, "Can't construct AllelicCount with negative read counts.");
+        /* these can be null */
+        this.refNucleotide = refNucleotide;
+        this.altNucleotide = altNucleotide;
+    }
+
+    public AllelicCount(final SimpleInterval interval, final int refReadCount, final int altReadCount) {
+        this(interval, refReadCount, altReadCount, null, null);
+    }
+
+    @Override
+    public String getContig() { return interval.getContig(); }
+
+    @Override
+    public int getStart() { return interval.getStart(); }
+
+    @Override
+    public int getEnd() { return interval.getEnd(); }
+
+    public SimpleInterval getInterval() { return interval; }
+
+    public int getRefReadCount() { return refReadCount; }
+
+    public int getAltReadCount() { return altReadCount; }
+
+    public Nucleotide getRefNucleotide() {
+        return Utils.nonNull(refNucleotide, "The ref nucleotide is not set.");
+    }
+
+    public Nucleotide getAltNucleotide() {
+        return Utils.nonNull(altNucleotide, "The alt nucleotide is not set.");
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof AllelicCount)) {
+            return false;
+        }
+
+        final AllelicCount count = (AllelicCount) o;
+        return interval.equals(count.interval)
+                && refReadCount == count.refReadCount && altReadCount == count.altReadCount
+                && (!((refNucleotide != null) && (count.refNucleotide != null)) || (refNucleotide == count.refNucleotide))
```

Comment:
If nucleotides are missing for either count, we don't use them in the equality check.  Added some doc to clarify and cleaned up boolean logic.